### PR TITLE
Added functionality to stop duplicate countries being added on CountryOfConsumptionFromNi

### PR DESCRIPTION
**Pull request by susan-badel**

### CODE REVIEW

File: app/controllers/CountryOfConsumptionFromNiController.scala [hunk 1]
```diff
@@ -35,12 +36,13 @@ class CountryOfConsumptionFromNiController @Inject()(
                                         view: CountryOfConsumptionFromNiView
                                     )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
 
-  private val form = formProvider()
   protected val controllerComponents: MessagesControllerComponents = cc
 
   def onPageLoad(mode: Mode, period: Period, index: Index): Action[AnyContent] = cc.authAndGetData(period) {
     implicit request =>
 
+      val form = formProvider(index, request.userAnswers.get(AllSalesFromNiQuery).getOrElse(Seq.empty).map(_.countryOfConsumption))
```
Comment (darbyo) [inline app/controllers/CountryOfConsumptionFromNiController.scala:44]: Can we space this over multiple lines and the form in the onSubmit

File: app/forms/CountryOfConsumptionFromNiFormProvider.scala [hunk 2]
```diff
@@ -18,15 +18,16 @@ package forms
...
 import javax.inject.Inject
 import forms.mappings.Mappings
-import models.Country
+import models.{Country, Index}
 import play.api.data.Form
 
 class CountryOfConsumptionFromNiFormProvider @Inject() extends Mappings {
 
-  def apply(): Form[Country] =
+  def apply(thisIndex: Index, existingAnswers: Seq[Country]): Form[Country] =
```
Comment (darbyo) [inline app/forms/CountryOfConsumptionFromNiFormProvider.scala:26]: Can we just call it index?

File: app/forms/mappings/Constraints.scala [hunk 3]
```diff
@@ -126,4 +126,17 @@ trait Constraints {
       case _ =>
         Invalid(errorKey)
     }
+
+  protected def notADuplicate[A](index: Index, existingAnswers: Seq[A], errorKey: String, args: Any*): Constraint[A] = {
```
Comment (darbyo) [inline app/forms/mappings/Constraints.scala:130]: Can we add some tests for this method

**Review by darbyo — APPROVED** [review]

File: app/controllers/CountryOfConsumptionFromNiController.scala (hunk 1 above)
Comment (susan-badel) [reply app/controllers/CountryOfConsumptionFromNiController.scala]: Done

File: app/forms/CountryOfConsumptionFromNiFormProvider.scala (hunk 2 above)
Comment (susan-badel) [reply app/forms/CountryOfConsumptionFromNiFormProvider.scala]: Done

File: app/forms/mappings/Constraints.scala (hunk 3 above)
Comment (susan-badel) [reply app/forms/mappings/Constraints.scala:130]: Done

**Review by darbyo — APPROVED** [review]

**Pull request merged**